telegram/ext/filters.py

```diff
@@ -359,39 +371,46 @@ def name(self) -> str:
 
     @name.setter
     def name(self, name: str) -> NoReturn:
-        raise RuntimeError('Cannot set name for XORFilter')
+        raise RuntimeError(f'Cannot set name for {self.__class__.__name__!r}')
 
 
-class _DiceEmoji(MessageFilter):
+class _Dice(MessageFilter):
     __slots__ = ('emoji', 'values')
 
-    def __init__(self, values: SLT[int] = None, emoji: str = None):
-        name = f"filters.DICE.{getattr(emoji, 'name', '')}" if emoji else 'filters.DICE'
+    def __init__(self, values: SLT[int] = None, emoji: DiceEmojiEnum = None):
+        super().__init__()
         self.emoji = emoji
         self.values = [values] if isinstance(values, int) else values
-        if self.values:
-            self.name = f"{name.title().replace('_', '')}({self.values})"  # CAP_SNAKE -> CamelCase
+        emoji_name = getattr(emoji, 'name', '')  # Can be e.g. BASKETBALL (see emoji enums)
```

Comment:
shouldn't `emoji.name` suffice now if `emoji` is annotated as enum? you can move this under the `if emoji` clause to exclude the `None` case. And you can move the `if self.value and emoji` clause under the `if emoji` clause :)